Sources/muterCore/MutationOperators/MutationOperator.swift

```diff
@@ -53,12 +53,16 @@ struct MutationOperator {
 }
 
 protocol PositionSpecificRewriter: CustomStringConvertible {
-    var positionToMutate: AbsolutePosition { get }
-    init(positionToMutate: AbsolutePosition)
-    func visit(_ token: Syntax) -> Syntax
+    var positionToMutate: MutationPosition { get }
+    
+    init(positionToMutate: MutationPosition)
+    
+    func visit(_ node: SourceFileSyntax) -> Syntax
 }
 
 protocol PositionDiscoveringVisitor {
-    var positionsOfToken: [AbsolutePosition] { get }
-    func visit(_ token: SourceFileSyntax)
+    var positionsOfToken: [MutationPosition] { get }
+    init(configuration: MuterConfiguration?, sourceFileInfo: SourceFileInfo)
```

Comment:
Why do you need to declare the init here? I couldn't find where you are using it.

Reply:
All the `visitors` have this function signature

Reply:
Ok, but it is considered a new line. So, where are you using this new API of this protocol that isn't needed before? Sorry if I am pushing it, but I really didn't find the need yet.

Reply:
Check `TokenAwareVisitor`, `RemoveSideEffectsOperator.Visitor` and `ExcludedMutationPointsDetector`

They all have the same `init`...what we did was to lift that up to the protocol

Reply:
But what is the motivation? Who creates those classes doesn't already know the concrete classes? The init declared on the protocol is required only when the initialization has some abstraction, isn't it?

Reply:
All visitors may need the configuration for one reason or another.
All visitors will need the path and source code (this is b/c in order to correctly calculate the position of a given node, swift syntax needs both)